mozci/console/commands/check_backfills.py

```diff
@@ -0,0 +1,173 @@
+                )
+                continue
+
+            for indexed_task in indexed_tasks:
+                task_id = indexed_task["taskId"]
+                try:
+                    children_tasks = list_dependent_tasks(task_id)
+                except requests.exceptions.HTTPError as e:
+                    self.line(
+                        f"<error>Couldn't fetch dependent tasks of indexed task {task_id} on push {push.push_uuid}: {e}</error>"
+                    )
+                    continue
+
+                for child_task in children_tasks:
+                    task_action = (
+                        child_task.get("task", {}).get("tags", {}).get("action", "")
+                    )
+                    # We are looking for the Treeherder symbol because Sheriffs are
+                    # only interested in backfill-tasks holding the '-bk' suffix in TH
+                    th_symbol = (
+                        child_task.get("task", {})
+                        .get("extra", {})
+                        .get("treeherder", {})
+                        .get("symbol", "")
+                    )
+                    status = child_task.get("status", {})
+                    if task_action == "backfill-task" and th_symbol.endswith("-bk"):
+                        assert status.get(
+                            "taskId"
+                        ), "Missing taskId attribute in backfill task status"
+                        assert status.get(
+                            "taskGroupId"
+                        ), "Missing taskGroupId attribute in backfill task status"
+                        assert status.get(
+                            "state"
+                        ), "Missing state attribute in backfill task status"
+                        backfill_tasks.append(
+                            BackfillTask(
+                                status["taskId"], status["taskGroupId"], status["state"]
+                            )
+                        )
+                    else:
+                        logger.debug(
+                            f"Skipping non-backfill task {status.get('taskId')}"
+                        )
+
+            def group_key(task):
+                return task.group_id
+
+            # Sorting backfill tasks by their Taskcluster backfill group_id
+            backfill_tasks = sorted(backfill_tasks, key=group_key)
+            # Grouping ordered backfill tasks by their associated Taskcluster backfill group
+            for group_id, value in groupby(backfill_tasks, group_key):
+                tasks = list(value)
+                if all(task.state == "completed" for task in tasks):
+                    index_path = f"project.mozci.check-backfill.{environment}.{push.branch}.{push.rev}.{group_id}"
+                    already_indexed = [
+                        indexed
+                        for indexed in list_indexed_tasks(
```

Comment:
Nit: In this case we should only ever have one indexed task, right? Maybe we can simply check for the existence of the index instead of listing?